optuna/integration/lightgbm.py

```diff
@@ -62,60 +63,85 @@ class LightGBMPruningCallback(object):
             Note that this argument will be ignored if you are calling
             `cv method <https://lightgbm.readthedocs.io/en/latest/Python-API.html#lightgbm.cv>`_
             instead of train method.
+        report_interval:
+            Check if the trial should be reported for pruning every n-th epoch.
+            By default ``report_interval=1`` and reporting is performed after every epoch.
+            Note that the pruning itself is performed according to the interval definition
+            of the pruner.
     """
 
     def __init__(
-        self, trial: optuna.trial.Trial, metric: str, valid_name: str = "valid_0"
+        self,
+        trial: optuna.trial.Trial,
+        metric: str,
+        valid_name: str = "valid_0",
+        report_interval: int = 1,
     ) -> None:
 
         _imports.check()
 
         self._trial = trial
         self._valid_name = valid_name
         self._metric = metric
+        self._report_interval = report_interval
 
-    def __call__(self, env: "CallbackEnv") -> None:
-
-        # If this callback has been passed to `lightgbm.cv` function,
-        # the value of `is_cv` becomes `True`. See also:
-        # https://github.com/Microsoft/LightGBM/blob/v2.2.2/python-package/lightgbm/engine.py#L329
-        # Note that `5` is not the number of folds but the length of sequence.
-        is_cv = len(env.evaluation_result_list) > 0 and len(env.evaluation_result_list[0]) == 5
-        if is_cv:
-            target_valid_name = "cv_agg"
-        else:
-            target_valid_name = self._valid_name
+    def _find_evaluation_result(
+        self, target_valid_name: str, env: "CallbackEnv"
+    ) -> Optional[list]:
 
         for evaluation_result in env.evaluation_result_list:
             valid_name, metric, current_score, is_higher_better = evaluation_result[:4]
             if valid_name != target_valid_name or metric != self._metric:
                 continue
 
+            return evaluation_result
+
+        return None
+
+    def __call__(self, env: "CallbackEnv") -> None:
+
+        if (env.iteration + 1) % self._report_interval == 0:
```

Comment:
A potential discussion point: `(env.iteration + 1) % self._report_interval` or `env.iteration % self._report_interval`

Reply:
I'd got with `+1` since that conforms to the expectation of every 5 trials, given the trials start at base 0.